### ISSUE TITLE
String incompatibility build error in VS2015

### ISSUE THREAD

**Issue by AndyDentFree** [realm/realm-object-store#139]
The declaration of `Applier::operator()(ChangePrimaryKey)` in `object_store.cpp` triggers the following error when compiling with VS:
`error C2280: 'realm::StringData::StringData<std::char_traits<char>,std::allocator<char>>(std::basic_string<char,std::char_traits<char>,std::allocator<char>> &&)': attempting to reference a deleted function`

I think the key thing is that properties being used are **std::string** but the signature to `ObjectStore::set_primary_key_for_object` takes `StringData` objects and the relevant constructors is declared directly using `std::basic_string`

```
template<class T, class A>
StringData(const std::basic_string<char, T, A>&);
```

It is quite happy if temporary `StringData()` objects are passed.

**Comment by bdash** [comment]
The line of code in question is (wrapped for readability in GitHub's narrow comment view):
```c++
ObjectStore::set_primary_key_for_object(group, op.object->name,
                                        op.property ? op.property->name : "");
```

The compiler error indicates that compiler is selecting a `StringData` constructor overload that was explicitly deleted, namely the constructor that converts from an rvalue `std::string`:
```c++
// StringData does not store data, callers must manage their own strings.
template<class T, class A>
StringData(std::basic_string<char, T, A>&&) = delete;
```

I'm not 100% clear on why this works for GCC and Clang, and whether it's expected to work per the C++ standard. I suspect @tgoyne may be able to provide an explanation here.

We can work around the MSVC issue by introducing a local `std::string` to hold the result of the ternary operator.

**Comment by bdash** [comment]
This was fixed by #143.
